### ISSUE TITLE
ERROR: dependency ‘arrangements’ is not available for package ‘SPOTlight’

### ISSUE THREAD

**Issue by galaxyeee** [MarcElosua/SPOTlight#81]
Dear SPOTlight developers,

Hello, I am writing an issue because I have a questions.
I tried to install the 0.1.7 version to apply the previous version of the tutorial, but it failed.
`devtools::install_github("https://github.com/MarcElosua/SPOTlight", ref = "spotlight-0.1.7")`
```
ERROR: dependency ‘arrangements’ is not available for package ‘SPOTlight’
* removing ‘/home/eunha/R/x86_64-pc-linux-gnu-library/4.2/SPOTlight’
Warning messages:
1: In i.p(...) : installation of package ‘gmp’ had non-zero exit status
2: In i.p(...) :
  installation of package ‘arrangements’ had non-zero exit status
3: In i.p(...) :
  installation of package ‘/tmp/Rtmp1GS8Q2/file3329765b4b89c/SPOTlight_0.1.7.tar.gz’ had non-zero exit status
```
So I tried to install using the source file, but couldn't find the source file.
If there is any other way, please let me know.

Thank for your help.

Best,
EUNHA LIM

**Comment by MarcElosua** [comment]
Hi @galaxyeee,

Thank you for your interest in SPOTlight! Have you tried installing the package `arrangements`? It seems like it is an unmet dependency!

Best,
Marc

**Comment by galaxyeee** [comment]
Hi @MarcElosua ,

Thank you very much for your kind reply. This issue has been resolved. 
However, the error was not resolved during troubleshooting, so I asked again.
This error occurred during the creation of "mgs_df".

`genes <- !grepl(pattern = "^Rp[l|s]|Mt", x = rownames(brca.sce))`
I converted the single cell seurat object as a reference into a SingleCellExperiment object and used that data to save the "genes".

`mgs <- scoreMarkers(brca.sce, subset.row = genes)`
After going through the scoremarkers function, this error occurred.
```
Error in FUN(x, table, nomatch = nomatch, incomparables = incomparables) : 
  'match' requires vector arguments

5. FUN(x, table, nomatch = nomatch, incomparables = incomparables)
4. match(desired.comparisons, averaged.comparisons)
3. match(desired.comparisons, averaged.comparisons)
2. .cross_reference_to_desired(pre.ave$averaged.comparisons, desired.comparisons, collapse.symmetric = collapse.symmetric)
1. scoreMarkers(brca.sce, row.data = genes, groups = brca.sce$celltype_major)
```

Or when using SPOTlight function, I wonder if I can just put the markerlist data frame in parameter "mgs" and use other values than "mean.AUC" in "weight_id".

I would really appreciate it if you let me know if there is a solution. Thank you!

Best,
EUNHA
